lib/new_relic/agent/instrumentation/grpc/server/instrumentation.rb

```diff
@@ -38,18 +39,23 @@ def add_http2_port_with_tracing(*args)
 
           def run_with_tracing(*args)
             set_host_and_port_and_method_info_on_desc
-
             yield
           end
 
           private
 
-          def set_host_and_port_on_server_instace(host_string)
+          def host_and_port_from_host_string(host_string)
+            return [nil, nil] unless host_string
+
             info = host_string.split(':')
 
-            # TODO:
-            # raise if info.size != 2
+            return [nil, nil] unless info.size == 2
```

Comment:
Since `[nil, nil]` is referenced twice, would it make sense to set it as a variable?

Reply:
Great question. The arrays are both hidden behind conditionals that we currently do not expect to evaluate as true very often. My hope was that the happy path would not bear the cost of allocating the object array at all. Switching to a constant would have all paths always defining the array.

But now that I look over this code again, I think we can refactor things to just perform a `return` (without params). For some reason I wanted to explicitly set the two instance vars even if they had `nil` as a value. But Ruby doesn't complain when you ask for a non-existent instance variable and just gives you back the same `nil`, so I think we can clean this up a bit. Will update!

Reply:
Addressed with [5f30f9e](https://github.com/newrelic/newrelic-ruby-agent/pull/1283/commits/5f30f9e0794726d7b2bad11287f7874c678a2fa5)